candidategenerator.js

```diff
@@ -28,14 +28,82 @@ var wNum = 0;
    - obeysExhaustivity (boolean or array of categories at which to require conformity to exhaustivity)
    - obeysHeadedness (boolean)
    - obeysNonrecursivity (boolean)
+	 - rootCategory (string)
+	 - recursiveCategory (string)
+	 - terminalCategory (string)
    - addTones (string). Possible values include:
 	 		- "addJapaneseTones"
 			- "addIrishTones_Elfner"
 			- "addIrishTones_Kalivoda"
+	- noUnary (boolean): if true, don't create any nodes that immediately dominate only a single terminal.
+	- requireRecWrapper (boolean). Formerly "requirePhiStem"
 */
 window.GEN = function(sTree, words, options){
 	options = options || {}; // if options is undefined, set it to an empty object (so you can query its properties without crashing things)
 
+	options.recursiveCategory = options.recursiveCategory || "phi"; //sets the default of recursiveCategory option to "phi"
+
+	//set default root root category based on options passed
+	if(!options.rootCategory){ //root category is unspecifed
+		//recursiveCategory is specified as "w"
+		if(options.recursiveCategory === "w"){
+			options.rootCategory = "phi";
+		}
+		else{
+			options.rootCategory = "i";
+		}
+	}
+
+	//set default terminal category based on options passed
+	if(!options.terminalCategory){ //terminalCategory is unspecified
+		//terminalCategory unspecified, recursiveCategory specified as "i"
+		if(options.recursiveCategory === "i"){
```

Comment:
Can we abstract this? Instead of using 3 conditionals, use pcat.nextLower(options.recursiveCategory)
Recursive category is guaranteed to be specified at this point since you set it in line 44.

Reply:
As it stands, the defaults for rootCategory and terminalCategory are based on the value of recursiveCategory. Is this the way we want it or do we want to do it some other way, eg. the defaults for recursiveCategory and terminalCategory could be based on rootCategory.

Reply:
Also: the default for recursiveCategory is "phi", do we want to make reference to the pCat instead? I feel like either way someone might get confused.

Reply:
These are good questions. I think we should stick with hard-coded 'phi' as the default, just in case pCat changes somehow. Also, defining the default root and terminal categories based on the recursive category seems good to me.